storage/diemdb/src/lib.rs

```diff
@@ -100,6 +138,63 @@ fn gen_rocksdb_options(config: &RocksdbConfig) -> Options {
     db_opts
 }
 
+#[derive(Debug)]
+struct RocksdbPropertyReporter {
+    sender: Option<Mutex<mpsc::Sender<bool>>>,
+    join_handle: Option<JoinHandle<()>>,
+}
+
+impl RocksdbPropertyReporter {
+    fn new(db: Arc<DB>) -> Self {
+        let (send, recv) = mpsc::channel();
+        let join_handle = Some(thread::spawn(move || loop {
+            if let Err(e) = Self::update_rocksdb_properties(&db) {
+                warn!(
+                    error = ?e,
+                    "update rocksdb property failure."
+                );
+            }
+            // report rocksdb properties each 10 second
+            if let Err(e) = recv.recv_timeout(Duration::from_secs(10)) {
+                match e {
+                    mpsc::RecvTimeoutError::Timeout => (),
+                    mpsc::RecvTimeoutError::Disconnected => break,
+                }
+            }
+        }));
+        Self {
+            sender: Some(Mutex::new(send)),
+            join_handle,
+        }
+    }
+
+    fn update_rocksdb_properties(db: &DB) -> Result<()> {
+        let _timer = DIEM_STORAGE_OTHER_TIMERS_SECONDS
+            .with_label_values(&["update_rocksdb_properties"])
+            .start_timer();
+        for cf_name in DiemDB::column_families() {
+            for (property_name, rocksdb_property_argument) in &*ROCKSDB_PROPERTY_MAP {
+                DIEM_STORAGE_ROCKSDB_PROPERTIES
+                    .with_label_values(&[cf_name, property_name])
+                    .set(db.get_property(cf_name, rocksdb_property_argument)? as i64);
+            }
+        }
+        Ok(())
+    }
+}
+
+impl Drop for RocksdbPropertyReporter {
+    fn drop(&mut self) {
+        // Drop the sender
+        self.sender.take();
```

Comment:
This is a little bit inexplicit.. Why not send a `()` to indicate "quit"?

Reply:
sg